### DISCUSSION TITLE
Shelly PlusPMMini: Tasmota shows always positive power and current values

### DISCUSSION THREAD

**Discussion by lsbbs** [arendst/Tasmota#20322] [Ideas]
### PROBLEM DESCRIPTION
Installed Tasmota 13.3.0 via OTA on two ShellyPMmini to get VA and VAr what the Shelly firmware does not.
Is working so far fine but on one meter I have power import and export. Unfortunate Tasmota is showing always positive values for the current. The Shelly firmware was showing import positive and export negative.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ x] Device used (e.g., Sonoff Basic): Shelly PlusPMMini
- [x ] Tasmota binary firmware version number used: 13.3.0
  - [x ] Pre-compiled
  - [ ] Self-compiled
- [x ] Flashing tools used: Firmare upload as .zip
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
13:47:17.334 MQT: usv-meter/usv-in/RESULT = {"NAME":"Shelly PlusPMMini","GPIO":[576,32,0,4736,0,0,3200,8161,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}
13:47:17.560 MQT: usv-meter/usv-in/RESULT = {"Module":{"0":"Shelly PlusPMMini"}}
13:47:17.762 MQT: usv-meter/usv-in/RESULT = {"GPIO0":{"576":"LedLink_i"},"GPIO1":{"32":"Button1"},"GPIO2":{"0":"None"},"GPIO3":{"4736":"ADC Temp1"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO6":{"3200":"Serial Tx"},"GPIO7":{"8161":"BL0942 Rx2"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO20":{"0":"None"},"GPIO21":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
13:47:17.334 MQT: usv-meter/usv-in/RESULT = {"NAME":"Shelly PlusPMMini","GPIO":[576,32,0,4736,0,0,3200,8161,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}
13:47:17.560 MQT: usv-meter/usv-in/RESULT = {"Module":{"0":"Shelly PlusPMMini"}}
13:47:17.762 MQT: usv-meter/usv-in/RESULT = {"GPIO0":{"576":"LedLink_i"},"GPIO1":{"32":"Button1"},"GPIO2":{"0":"None"},"GPIO3":{"4736":"ADC Temp1"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO6":{"3200":"Serial Tx"},"GPIO7":{"8161":"BL0942 Rx2"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO20":{"0":"None"},"GPIO21":{"0":"None"}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

Changing current direction the values remain positive.

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

Current going from the L to the O terminal should be shown as positive value.
Current going from the O to the L terminal should be shown as negative value.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by arendst** [comment]
Pls provide some logging information when export is active. Set ``weblog 4`` for a minute and provide the log like:
```
18:01:17.851 CMD: weblog 4
18:01:17.852 SRC: WebConsole from 0.0.0.0
18:01:17.853 CMD: Grp 0, Cmd 'WEBLOG', Idx 1, Len 1, Pld 4, Data '4'
18:01:17.856 MQT: stat/plus1pmmini/RESULT = {"WebLog":4}
18:01:17.983 BL9: Rx 55000000AEB430560B000000000000002E4E00220100C0
18:01:18.883 CFG: Saved, Count 495, Bytes 4096
18:01:18.973 BL9: Rx 55000000658130710700000000000000364E000201003D
18:01:19.973 BL9: Rx 55000000F31830E70900000000000000364E00020100A0
18:01:20.980 BL9: Rx 55000000E73B30A00E00000000000000364E00020100CB
18:01:21.978 BL9: Rx 55110000DEC7307F0B00010000000000364E000201005A
18:01:22.982 BL9: Rx 550000001120314F0B00000000000000364E00220100EF
18:01:23.984 BL9: Rx 5500000040BE30450900000000000000364E002201002F
18:01:24.989 BL9: Rx 55000000A43130370B00000000000000364E0022010064
18:01:25.989 BL9: Rx 55000000EF23308A07000000000000003E4E00220100D0
18:01:26.994 BL9: Rx 55000000EA7E309C0A00000000000000364E000201008D
18:01:27.994 BL9: Rx 55000000E8E030080B00000000000000364E00020100C0
18:01:29.000 BL9: Rx 55000000DFBF30160A00000000000000364E00020100DD
18:01:29.018 WIF: Checking connection...
18:01:29.967 BL9: Rx 55000000CEC530DA0800000000000000364E0022010006
18:01:31.005 BL9: Rx 55000000B17D307807000000000000003E4E00220100C6
18:01:32.005 BL9: Rx 55000000487830DA0B000000000000003E4E00220100CE
18:01:33.013 BL9: Rx 550000007F19300A0C000000000000003E4E00220100C5
18:01:34.014 BL9: Rx 55000000C2B230E40B00000000000000364E0022010018
18:01:34.935 CMD: weblog 3
```

**Comment by arendst** [comment]
BTW This is the reason why you won't see negative numbers https://github.com/arendst/Tasmota/issues/15374#issuecomment-1105293179

**Comment by arendst** [comment]
Install latest dev branch and change in template GPIO07 index of BL0942 Rx from 2 to 6

**Reply by lsbbs** [reply]
Just compiled and installed it.
Active Power Export  is now negative.

Apparent Power, Reactive Power still positive.
Power Factor is negative.

![Screenshot_20231230_111122](https://github.com/arendst/Tasmota/assets/7830198/47eaae13-9377-41c7-9313-62b5856e6e2d)

**Reply by Jason2866** [reply]
Is the power factor that horrible too when "consuming" energy? 
A power factor of 0.1 should not happen.
Is the device calibrated after Tasmota was installed?

**Reply by lsbbs** [reply]
This measurements are wrong.
The power factor is 0.95 to 1
But for the moment I am happy with correct active power.

**Reply by Jason2866** [reply]
Have you fully calibrated the device? Without you will get wrong values.

**Reply by lsbbs** [reply]
In this special case I made this screenshot a power factor of 0.1 is possible.
On the output of this inverter-charger a solar inverter is connected. From there the power export is coming. And the charger is recharging batteries.
For the Shelly at this moment is visible a 500VA transformer and a very little amount of power coming out.
When the power export is rising and the charger stopping the cos phi is going up to around 0.98
